src/components/ha-hls-player.ts

```diff
@@ -0,0 +1,198 @@
+import {
+  css,
+  CSSResult,
+  customElement,
+  html,
+  LitElement,
+  property,
+  PropertyValues,
+  query,
+  TemplateResult,
+} from "lit-element";
+import { fireEvent } from "../common/dom/fire_event";
+import { nextRender } from "../common/util/render-status";
+import { getExternalConfig } from "../external_app/external_config";
+import type { HomeAssistant } from "../types";
+
+type HLSModule = typeof import("hls.js");
+
+@customElement("ha-hls-player")
+class HaHLSPlayer extends LitElement {
+  @property({ attribute: false }) public hass!: HomeAssistant;
+
+  @property() public url!: string;
+
+  @property({ type: Boolean, attribute: "controls" })
+  public controls = false;
+
+  @property({ type: Boolean, attribute: "muted" })
+  public muted = false;
+
+  @property({ type: Boolean, attribute: "autoplay" })
+  public autoPlay = false;
+
+  @property({ type: Boolean, attribute: "playsinline" })
+  public playsInline = false;
+
+  @query("video") private _videoEl!: HTMLVideoElement;
+
+  private _hlsPolyfillInstance?: Hls;
+
+  private _useExoPlayer = false;
+
+  public connectedCallback() {
+    super.connectedCallback();
+    this._startHls();
```

Comment:
We will now also do this on first connected, and only after being disconnected. Copy the behavior of camera-stream, with the `this._attached` property and handle it all in updated.